eip7594/src/lib.rs

```diff
@@ -86,29 +95,37 @@ impl From<ThreadCount> for usize {
 /// The context that will be used to create and verify opening proofs.
 #[derive(Debug)]
 pub struct DASContext {
+    #[cfg(feature = "multithreading")]
     thread_pool: Arc<ThreadPool>,
     pub prover_ctx: ProverContext,
     pub verifier_ctx: VerifierContext,
 }
 
+#[cfg(feature = "multithreading")]
 impl Default for DASContext {
     fn default() -> Self {
         let trusted_setup = TrustedSetup::default();
         const DEFAULT_NUM_THREADS: ThreadCount = ThreadCount::Single;
```

Comment:
Might make sense to make the default `SensibleDefault` and then have that use the global threadpool

Reply:
In terms of defaults, single-thread, global threadpool and then a feature flag for local threadpool with threadcount might be the best, or just ask users to import threadpool in their own crate and set the threadpool amount themselves